bitnami/redis/values.yaml

```diff
@@ -494,7 +494,14 @@ master:
     ## e.g.
     ## loadBalancerSourceRanges:
     ##   - 10.10.10.0/24
-    ##
+    ##    
+    externalIPs: []
+    ## @param master.service.externalIPs Redis&reg; master service External IPs
+    ## https://kubernetes.io/docs/concepts/services-networking/service/#external-ips
+    ## e.g.
+    ## externalIPs:
```

Comment:
I have tried to set the externalIPs with the following value:
```
    externalIPs:
      - 10.10.10.1
```
And I got this error:
`Error: INSTALLATION FAILED: Service "sp-red-redis-master" is invalid: spec.sessionAffinity: Unsupported value: "None - 10.202.13.2": supported values: "ClientIP", "None"`
Should I configure something else to test it?

Reply:
Hello, @corico44 
Could you send me the yaml generated by helm?
Your service.yaml has to have at least this parameters to work:
```yaml
apiVersion: v1
kind: Service
metadata:
  name: dev-app0-svc-teste
spec:
  selector:
    app.kubernetes.io/component: master
    app.kubernetes.io/instance: dev-app01
    app.kubernetes.io/name: redis
  ports:
    - name: tcp-redis
      protocol: TCP
      port: 29001
      targetPort: redis
  externalIPs:
    - 10.10.10.1
  type: ClusterIP
 ```

Reply:
I am only modifying the parameter in the values.yaml adding the ip and I execute the following command: 
`helm install sp-red . -f values.yaml`
which gives me the indicated error. So I can't access any generated yaml because it doesn't get installed.

Reply:
Hello @corico44 

I made another commit to solve this and test it.

Here the how to:

create a file : values-test.yml
```yaml
architecture: standalone
auth:
  password: bla321
image:
  registry: registry.prodest.es.gov.br
  repository: docker/bitnami/redis
master:
  containerSecurityContext:
    runAsUser: 1000750001
  livenessProbe:
    initialDelaySeconds: '10'
  persistence:
    accessModes:
      - ReadWriteMany
    size: 1Gi
  podSecurityContext:
    enabled: false
  readinessProbe:
    initialDelaySeconds: '10'
  service:
    externalIPs:
      - 10.10.10.1
    ports:
      redis: 29001
```

Follow theses steps
```bash
git clone https://github.com/dboc/charts.git

minikube start --driver=docker
minikube addons enable ingress
# Verify all manifest with --dry-run --debug
helm dependency update charts/bitnami/redis
helm install myredis --dry-run --debug -f values-test.yml ./charts/bitnami/redis
# Install helm chart redis
helm install myredis -f values-test.yml ./charts/bitnami/redis
# Verify install
kubectl get pods -A # waiting until Running state
kubectl get svc -A
# Output has to be like that
#     NAMESPACE     NAME               TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                  AGE
#     default       kubernetes         ClusterIP   10.96.0.1       <none>        443/TCP                  3m18s
#     default       myredis-headless   ClusterIP   None            <none>        29001/TCP                48s
#**** default       myredis-master     ClusterIP   10.103.122.14   10.10.10.1    29001/TCP                48s <<<<<<<<<< HERE ALL EXTERNAL-IP
#     kube-system   kube-dns           ClusterIP   10.96.0.10      <none>        53/UDP,53/TCP,9153/TCP   3m17s
# testing just the redis
kubectl port-forward --namespace default svc/myredis-master 29001:29001 &
telnet 127.0.0.1 29001
> auth bla321
> quit now
# for testing the external IP you have to make sure the IP is routing to at least one node in the cluster kubernetes
# with this premisse, just do it: telnet externalIP 29001

# Testing chart without use ExternalIP - empty values
helm install myredis01  ./charts/bitnami/redis
# Verify install
kubectl get pods -A # waiting until Running state
kubectl get svc -A
#     NAMESPACE     NAME                 TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                  AGE
#     default       kubernetes           ClusterIP   10.96.0.1        <none>        443/TCP                  10m
#     default       myredis-headless     ClusterIP   None             <none>        29001/TCP                7m53s
#     default       myredis-master       ClusterIP   10.103.122.14    10.10.10.1    29001/TCP                7m53s
#     default       myredis01-headless   ClusterIP   None             <none>        6379/TCP                 26s
#**** default       myredis01-master     ClusterIP   10.103.121.11    <none>        6379/TCP                 26s <<<<<<<<<< HERE Without EXTERNAL-IP
#     default       myredis01-replicas   ClusterIP   10.106.124.243   <none>        6379/TCP                 26s
#     kube-system   kube-dns             ClusterIP   10.96.0.10       <none>        53/UDP,53/TCP,9153/TCP   10m
kubectl port-forward --namespace default svc/myredis01-master 6379:6379 &
# testing just the redis
telnet 127.0.0.1 6379

# remove minikube
minikube stop
minikube delete
```

Now it working, please verify